salt/modules/netbox.py

```diff
@@ -117,10 +187,980 @@ def get(app, endpoint, id=None, **kwargs):
+        if not nb_role:
+            return False
+
+        nb_type = get_('dcim', 'device-types', model=model)
+        if not nb_type:
+            return False
+        nb_site = get_('dcim', 'sites', name=site)
+        if not nb_site:
+            return False
+
+        status = {'label': "Active", 'value': 1}
+    except RequestError as e:
+        log.error("{}, {}, {}".format(e.req.request.headers, e.request_body, e.error))
+        return False
+
+    payload = {'name': name, 'display_name': name, 'slug': slugify(name), 'device_type': nb_type['id'],
+               'device_role': nb_role['id'], 'site': nb_site['id']}
+    new_dev = _add('dcim', 'devices', payload)
+    if new_dev:
+        return {'dcim': {'devices': payload}}
+    else:
+        return False
+
+
+def update_device(name, **kwargs):
+    '''
+    .. versionadded:: Fluorine
+
+    Add attributes to an existing device, identified by name.
+
+    name
+        The name of the device, e.g., ``edge_router``
+    kwargs
+       Arguments to change in device, e.g., ``serial=JN2932930``
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion netbox.update_device edge_router serial=JN2932920
+    '''
+    kwargs = salt.utils.args.clean_kwargs(**kwargs)
+    nb_device = _get('dcim', 'devices', auth_required=True, name=name)
+    for k, v in kwargs.items():
+        setattr(nb_device, k, v)
+    try:
+        nb_device.save()
+        return {'dcim': {'devices': kwargs}}
+    except RequestError as e:
+        log.error("{}, {}, {}".format(e.req.request.headers, e.request_body, e.error))
+        return False
+
+
+def create_inventory_item(device_name,
+                          item_name,
+                          manufacturer_name=None,
+                          serial=None,
+                          part_id=None,
+                          description=None):
+    '''
+    .. versionadded:: Fluorine
+
+    Add an inventory item to an existing device.
+
+    device_name
+        The name of the device, e.g., ``edge_router``.
+    item_name
+        String of inventory item name, e.g., ``Transceiver``.
+
+    manufacturer_name
+        String of inventory item manufacturer, e.g., ``Fiberstore``.
+
+    serial
+        String of inventory item serial, e.g., ``FS1238931``.
+
+    part_id
+        String of inventory item part id, e.g., ``740-01234``.
+
+    description
+        String of inventory item description, e.g., ``SFP+-10G-LR``.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion netbox.create_inventory_item edge_router Transceiver part_id=740-01234
+    '''
+    nb_device = get_('dcim', 'devices', name=device_name)
+    if not nb_device:
+        return False
+    if manufacturer_name:
+        nb_man = get_('dcim', 'manufacturers', name=manufacturer_name)
+        if not nb_man:
+            create_manufacturer(manufacturer_name)
+            nb_man = get_('dcim', 'manufacturers', name=manufacturer_name)
+    if not serial:
```

Comment:
Should we just make the empty strings the keyword defaults?

Reply:
Done